pkg/util/cluster/aad.go

```diff
@@ -89,22 +74,29 @@ func (c *Cluster) createServicePrincipal(ctx context.Context, appID string) (str
 		return "", err
 	}
 
-	return *sp.ObjectID, nil
+	return *result.GetId(), nil
 }
 
 func (c *Cluster) deleteApplication(ctx context.Context, appID string) error {
```

Comment:
This function duplicates a lot from what's already in `GetServicePrincipalIDByAppID()`, other than the delete by application ID that happens if the result is successful. Can we consolidate somehow, or use that helper function to do the `Get` here instead of duplicating the logic?

Reply:
They're similar in principal, but they're operating on different API endpoints.  This function is calling the `Applications` endpoint whereas `GetServicePrincipalIDByAppID` is calling the `ServicePrincipals` endpoint.  And of course each endpoint has to define its own incompatible request structs, so there's actually not much commonality here.